src/duration.rs

```diff
@@ -498,16 +510,42 @@ impl Duration {
     }
 
     fn parse_time(bytes: &[u8], offset: usize, config: &TimeConfig) -> Result<Self, ParseError> {
-        let t = crate::time::PureTime::parse(bytes, offset, config)?;
+        let byte_len = bytes.len();
+        const HOUR_NUMERIC_LIMIT: i64 = 24 * 10i64.pow(8);
+        let mut hour: i64 = 0;
+        let mut day: u32 = 0;
+        let first_colon_index = bytes.iter().position(|&x| x == b':').unwrap();
 
-        if bytes.len() > t.position {
+        for idx in offset..first_colon_index {
+            let h = bytes.get(idx).ok_or(ParseError::InvalidCharHour)? - b'0';
+            if h > 9 {
+                return Err(ParseError::InvalidCharHour);
+            }
+            hour = (hour * 10) + (h as i64);
+        }
+        if hour > HOUR_NUMERIC_LIMIT {
+            return Err(ParseError::DurationHourValueTooLarge);
+        }
+
+        if byte_len - offset < 5 {
+            return Err(ParseError::TooShort);
+        }
+        let mut temp_vec: Vec<u8> = vec![b'0', b'0'];
+        temp_vec.extend_from_slice(&bytes[first_colon_index..]);
+        let new_bytes = &temp_vec[..];
```

Comment:
@davidhewitt actually there's also a case with fraction of second (microsecond) so that why I just want to set the hour part as `00`.

Reply:
Understood. Does it work if we generalize my suggestion to `b"00:00:00.000000"` instead?

Reply:
I don't know but the fraction part will always have to have 6 chars ? if so then yes.

Reply:
Because we slice it with `&mut new_bytes[... 3 + remaining.len()]` then we won't actually use any of the zeros beyond what's actually in the `remaining` data.

And I _think_ our limit is microseconds (i.e. max 6 chars). So this should be ok, I think.

Reply:
@davidhewitt ok. I made the change but had a trouble with a test `test_duration_parse_truncate_seconds`. Can you help me with that. It's fraction issue.

Reply:
Sure thing, https://github.com/pydantic/speedate/pull/64#discussion_r1648800346